### PR TITLE
Add new @vocabulary and @vocabulary-uri attributes for elements relating to aid-type

### LINKED ISSUE #346: Add new @vocabulary attributes for elements relating to aid-type
Add a new attribute at the following locations:

- `iati-activity/default-aid-type/@vocabulary`
- `iati-activity/transaction/aid-type/@vocabulary`

...with definition:

> A code for the vocabulary aid-type classifications. If omitted the AidType (OECD DAC) codelist is assumed. The code must be a valid value in the AidTypeVocabulary codelist

The attribute will have type `xsd:string` with occurrence rule `0..1`.

**Tasks:**

- [x] Add tests
- [x] Add `iati-activities/iati-activity/default-aid-type/@vocabulary`
- [x] Add `iati-activity/transaction/aid-type/@vocabulary`
- [x] Update Codelist mapping file: `IATI-Codelists/mapping.xml` (to link this attribute to the `AidTypeVocabulary` codelist)
- [x] `AidTypeVocabulary` codelist added to `IATI-Codelists-NonEmbedded` (https://github.com/IATI/IATI-Codelists-NonEmbedded/pull/188)

### PR DESCRIPTION
**Pull request by allthatilk**
Fixes #346 for schema

### CODE REVIEW

File: iati-activities-schema.xsd [hunk 1]
```diff
@@ -2804,5 +2806,13 @@
     </xsd:annotation>
   </xsd:attribute>
 
+  <xsd:attribute name="vocabulary" type="xsd:string">
```
Comment (dalepotter) [inline iati-activities-schema.xsd]: `vocabulary` is a commonly-used term, so would suggest naming this `aidTypeVocabulary`.

File: tests/should-pass/iati-activities/iati-activity/default-aid-type/_vocabulary/attribute-exists.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0"?>
+
+<iati-activities version="xx">
+  <iati-activity>
+    <iati-identifier></iati-identifier>
+    <reporting-org type="xx" ref="xx"><narrative>Organisation name</narrative></reporting-org>
```
Comment (dalepotter) [inline tests/should-pass/iati-activities/iati-activity/default-aid-type/_vocabulary/attribute-exists.xml:6]: For info, I've added an issue to fix the formatting of this line en masse before release: https://github.com/IATI/IATI-Schemas/issues/403

**Review by dalepotter — CHANGES_REQUESTED** [review]
One small thing re naming of the attribute.

Otherwise looks good :1st_place_medal:

**Comment by allthatilk** [comment]
@dalepotter ready for rereview

File: iati-activities-schema.xsd [hunk 3]
```diff
@@ -2806,13 +2778,25 @@
...
 
-  <xsd:attribute name="vocabulary" type="xsd:string">
-    <xsd:annotation>
-      <xsd:documentation xml:lang="en">
-        A code for the vocabulary aid-type classifications. If omitted the AidType (OECD DAC) codelist is assumed. The code must be a valid value in the AidTypeVocabulary codelist.
-      </xsd:documentation>
-    </xsd:annotation>
-  </xsd:attribute>
-
+  <xsd:complexType name="aid-type-base">
```
Comment (dalepotter) [inline iati-activities-schema.xsd]: I think we tend to name `complexType`s using lower camel case  - e.g. in https://github.com/IATI/IATI-Schemas/pull/372/files#diff-644a12e8a6ade95722e1f3adf60b7462R148

**Review by dalepotter — CHANGES_REQUESTED** [review]
One thing on the naming of the `complexType`.

**Comment by allthatilk** [comment]
@dalepotter ready for rerereview

**Review by dalepotter — APPROVED** [review]
Looks good!

**Pull request merged**